### DISCUSSION TITLE
$el of Alpine.Data()

### DISCUSSION THREAD

**Discussion by SirMetathyst** [alpinejs/alpine#2975] [1. Help] [answered]
Hello. Is there a way I can obtain the $el of current element the ``Alpine.data()`` component is attached to or do I have to explicitly pass in $el?

**Comment by SimoTod** [comment, marked as answer]
`this.$el`
https://codepen.io/SimoTod/pen/VwQBoLM?editors=1111

**Reply by SirMetathyst** [reply]
Thank you for your help. Tried a few ways, didn't realise I could only access it from a function (Javascript isn't a language I usually work in)

**Reply by SimoTod** [reply]
It's more of an Alpine thing, alpine.data is a factory. Outside a function, it would be evaluated immediately but at that point it doesn't have a clue about the element that is going to use the property.
